### PR TITLE
Add BUILD.gn for benchmarks

### PR DESCRIPTION
**Pull request by GaryQian**

### CODE REVIEW

File: build/secondary/testing/benchmark/BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  if (current_toolchain == host_toolchain) {
+    defines = [ "HAVE_STD_REGEX" ]
+  }
+
+  include_dirs = [
+    "//third_party/benchmark/include",
+  ]
+
+  sources = [
+    "//third_party/benchmark/src/arraysize.h",
```
Comment (chinmaygarde) [inline build/secondary/testing/benchmark/BUILD.gn]: I don't think you need the //third_party/benchmark prefix.

File: build/secondary/testing/benchmark/BUILD.gn [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+static_library("benchmark") {
+  if (current_toolchain == host_toolchain) {
+    defines = [ "HAVE_STD_REGEX" ]
```
Comment (chinmaygarde) [inline build/secondary/testing/benchmark/BUILD.gn]: Why is this required only for the host toolchain?

File: build/secondary/testing/benchmark/BUILD.gn (hunk 1 above)
Comment (GaryQian) [inline build/secondary/testing/benchmark/BUILD.gn]: This file is not in the //third_party/benchmark directory. Is there another way to reference those files without the prefix?
